### PR TITLE
gh-83004: Harden winreg init

### PR DESCRIPTION
**Pull request by erlend-aasland**
* Issue: gh-83004

### CODE REVIEW

**Comment by erlend-aasland** [comment]
@zooba, I'm landing this later tonight.

File: PC/winreg.c [hunk 1]
```diff
@@ -2170,7 +2172,14 @@ PyMODINIT_FUNC PyInit_winreg(void)
     ADD_INT(REG_RESOURCE_LIST);
     ADD_INT(REG_FULL_RESOURCE_DESCRIPTOR);
     ADD_INT(REG_RESOURCE_REQUIREMENTS_LIST);
+
+#undef ADD_INT
```
Comment (aisk) [inline PC/winreg.c:2176]: Should we undef the `ADD_KEY` here also?

Comment (erlend-aasland) [reply PC/winreg.c:2176]: Yes ideally, but not worth a separate PR, IMO.

Comment (aisk) [reply PC/winreg.c:2176]: Trying to merge this to #103250 , I'll add it is this PR.

**Pull request merged**